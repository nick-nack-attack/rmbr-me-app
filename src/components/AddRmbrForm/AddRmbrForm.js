import React, { Component } from 'react';
import { Input, Button } from '../Utils/Utils';
import RmbrApiService from "../../services/rmbr-api-service";
import RmbrmeContext from "../../contexts/RmbrmeContext";

export default class AddRmbrForm extends Component {

    constructor(props) {
        super(props)
        this.state = {
            rmbrTitle: '',
            error: null,
            onAddRmbrSuccess: () => {}
        }
    };

    static contextType = RmbrmeContext;

    handleTitleChange = changeEvent => {
        this.setState({
            rmbrTitle: changeEvent.target.value
        })
    };


    handleSubmitRmbr = e => {
        // this.context.addRmbr
        e.preventDefault()
        this.setState({error: null});
        const rmbr_title = this.state.rmbrTitle;
        const rmbr_text = ''
        const person_id = this.props.person_id;
        const user_id = window.localStorage.getItem('user_id');
        const category = 'Past';
        const newRmbr = { rmbr_title, category, rmbr_text, person_id, user_id };
        RmbrApiService.postRmbr(newRmbr)
            .then(() => {
                this.context.addRmbr(newRmbr)
                this.setState({
                    rmbrTitle: ''
                })
                this.props.onAddRmbrSuccess(newRmbr)
            })
            .catch(res => {
                this.setState({
                    error:res.error
                })
            })
        };

    render() {
        return (
            <form
                onSubmit={(e) => this.handleSubmitRmbr(e)}
            >
                <legend>Add Rmbr</legend>
                <div>
                   <Input
                        name='rmbrTitle'
                        id='rmbrTitle'
                        placeholder='Add a new rmbr here'
                        onChange={this.handleTitleChange}

                    />
                    <Button>Submit</Button>
                </div>
            </form>
        )
    }

}