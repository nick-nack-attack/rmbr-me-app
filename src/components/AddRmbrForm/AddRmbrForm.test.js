import React from "react";
import { shallow } from "enzyme";
import toJson from 'enzyme-to-json';
import AddRmbrForm from "./AddRmbrForm";

describe(`AddRmbrForm component`, () => {
    it(`renders the complete form`, () => {
        const wrapper = shallow(<AddRmbrForm/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});