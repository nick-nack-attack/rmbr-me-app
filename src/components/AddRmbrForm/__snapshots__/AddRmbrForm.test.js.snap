// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRmbrForm component renders the complete form 1`] = `
<form
  className="add_rmbr_form"
  onSubmit={[Function]}
>
  <legend>
    Add Rmbr
  </legend>
  <div>
    <Input
      id="rmbrTitle"
      name="rmbrTitle"
      onChange={[Function]}
      placeholder="Add a new rmbr here"
      value=""
    />
    <Button
      disabled={true}
    >
      Submit
    </Button>
  </div>
</form>
`;
